plugin/ipam/cni.go

```diff
@@ -35,31 +36,76 @@ func (i *Ipam) Allocate(args *skel.CmdArgs) (types.Result, error) {
 	if containerID == "" {
 		return nil, fmt.Errorf("Weave CNI Allocate: blank container name")
 	}
-	var ipnet *net.IPNet
 
-	if conf.Subnet == "" {
-		ipnet, err = i.weave.AllocateIP(containerID, false)
+	var ipconfigs []*current.IPConfig
+
+	if len(conf.IPs) > 0 {
```

Comment:
@bboreham I am a bit confused with how to support >1 IPs passed to the CNI via a `NetworkConfigList`, since the signature of `AddNetworkList` is:

```
func (c *CNIConfig) AddNetworkList(ctx context.Context, list *NetworkConfigList, rt *RuntimeConf) (types.Result, error) {
```

and `RuntimeConf` includes:
```
type RuntimeConf struct {
	ContainerID string
	NetNS       string
	IfName      string
...
```

Can a given interfaces (i.e. `eth1`) have multiple IP addresses? I am assuming we set `IfName` to `eth1` for example.

I hope that makes sense.

Reply:
Yes, an interface can have many IP addresses.  These days one IPv4 and one IPv6 is quite common (though not in Weave Net).

What you did is OK, though the Weave Net CNI plugin will currently only use the first address: https://github.com/weaveworks/weave/blob/b04217d8e6f252815016e226e21007c2076e6f69/plugin/net/cni.go#L85-L86